### PR TITLE
Improve MainView

### PR DESCRIPTION
**Pull request by chaixshot**
* Improve calculation center and zoom of MinimapPanel:InitializeMapSize()
* [Added number parameter to MinimapRaceCheckpoint.New()](https://github.com/manups4e/ScaleformUI/pull/239/commits/6ddc0894b6ced97d0b10e112a365c24816eb0574)
* Improved MainView:ProcessControl() holding behavior
* Optimized function math.round()
* Main loop optimized

Note:
* Please recheck C# syntax.
* White blip in picture it just a debugger

Screenshot 2024-10-26 181522
Screenshot 2024-10-26 181347
image

### CODE REVIEW

**Comment by manups4e** [comment]
ooooh awesome!!!

File: ScaleformUI_Lua/src/Menus/PauseMenus/LobbyMenu/MainView.lua [hunk 1]
```diff
@@ -562,38 +562,63 @@ function MainView:ProcessControl()
         return
     end
 
-    if (IsControlPressed(2, 172)) then
+    if (IsDisabledControlJustPressed(2, 172)) then
```
Comment (manups4e) [inline ScaleformUI_Lua/src/Menus/PauseMenus/LobbyMenu/MainView.lua:565]: why "just" pressed? the ControlPressed is done to allow the "keep button pressed to scroll"

Comment (chaixshot) [reply ScaleformUI_Lua/src/Menus/PauseMenus/LobbyMenu/MainView.lua:565]: This line check only client started press arrow up.
And keep repeat client holding arrow up on line 570

**Comment by manups4e** [comment]
Let me know when you're done and I'll merge this awesome pearl!🥰

**Comment by chaixshot** [comment]
I think i done with this now.

**Comment by manups4e** [comment]
i love this results!! thanks <3

**Pull request merged**